mesh-design.md

```diff
@@ -0,0 +1,270 @@
+	their descriptions are returned in the generated actor patches.
+
+Or at runtime:
+
+3. Through an Actor instance's `setCollider(colliderType: 'box' | 'sphere', ...moreOptions)` method, which takes
+	the shape, size, and options of the desired collider, and adds it to the accumulating actor patch. Next
+	`actor-update`, that collider patch is converted into a Unity collider of the appropriate type.
+
+**Note**: Mesh colliders in Unity are [expensive to generate, expensive to evaluate, and have a laundry list of
+caveats](https://docs.unity3d.com/Manual/class-MeshCollider.html). For this reason they generally advise against
+adding mesh colliders at all at runtime. So if we want MREs to have the ability to produce mesh colliders at all,
+we should be prepared for those performance penalties. For this reason, only `CreateFromGltf` can generate mesh
+colliders, though I suspect this isn't great either.
+
+**Note**: `Actor.CreatePrimitive` and `Actor.CreateFromGltf` are largely redundant with the introduction of assignable
+meshes, and may interfere with the functioning of mesh assignment without careful engineering. The following
+collider API proposal is designed primarily for use with `Actor.CreateEmpty`. See
+[Other Considerations](#removing-createprimitive-and-createfromgltf).
+
+Actor-creation-time collider initialization (1 and 2) is very convenient, because most of the time you don't
+need to change colliders at runtime. The challenge is that with mesh management, the mesh may not yet be loaded
+when the actor is created, or the mesh may be assigned well after actor creation. To match the convenience of the
+old mesh actor creation methods, I propose two changes.
+
+First, the creation of a new collider type: `auto`. When a mesh is loaded, we will register that mesh's preferred
+collider type based on the creation method, so primitives will get basic colliders (box, sphere, etc.) and glTF
+meshes will get either a bounding box or a mesh collider depending on `AssetContainer.loadGltf`'s `colliderType`
+argument (which matches the `CreateFromGltf` options). Then when a mesh is assigned to an actor with an `auto`
+collider type, a new collider is created on the actor based on the registered collider description for that mesh.
+
+Second, allow collider descriptions to be specified explicitly on initialization in the actor patch, instead of
+being a separate function argument like in `CreateFromGltf` or `CreatePrimitive`. This approach was rejected
+initially to prevent the user from specifying a mesh collider, but that could be done in other ways without
+complicating the API. See [Other Considerations](#consolidating-the-collider-apis-into-actor-patch).
+
+For example, if I wanted to create a box button, I might do something like this:
+
+```ts
+const context: MRE.Context;
+const assets: MRE.AssetContainer;
+
+// With a mesh asset and auto collider:
+const actor = MRE.Actor.CreateEmpty(context, {
```

Comment:
The name `Actor.CreateEmpty` doesn't work anymore. How about just `Actor.Create`?